pkg/services/accesscontrol/ossaccesscontrol/ossaccesscontrol.go

```diff
@@ -0,0 +1,64 @@
+package ossaccesscontrol
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/accesscontrol/evaluator"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// OSSAccessControlService is the service implementing role based access control.
+type OSSAccessControlService struct {
+	Cfg               *setting.Cfg `inject:""`
+	Log               log.Logger
+	builtInRoleGrants map[string][]string
+}
+
+func init() {
+	registry.RegisterService(&OSSAccessControlService{})
+}
+
+// Init initializes the OSSAccessControlService.
+func (ac *OSSAccessControlService) Init() error {
+	ac.Log = log.New("accesscontrol")
+	ac.builtInRoleGrants = builtInRoleGrants
+
+	return nil
+}
+
+func (ac *OSSAccessControlService) IsDisabled() bool {
+	if ac.Cfg == nil {
+		return true
+	}
+
+	_, exists := ac.Cfg.FeatureToggles["accesscontrol"]
+	return !exists
+}
+
+func (ac *OSSAccessControlService) Evaluate(ctx context.Context, user *models.SignedInUser, permission string, scope ...string) (bool, error) {
+	return evaluator.Evaluate(ac, ctx, user, permission, scope...)
+}
+
+func (ac *OSSAccessControlService) GetUserPermissions(ctx context.Context, user *models.SignedInUser, roles []string) ([]*accesscontrol.Permission, error) {
+	permissions := make([]*accesscontrol.Permission, 0)
+	for _, legacyRole := range roles {
+		if builtInRoleNames, ok := ac.builtInRoleGrants[legacyRole]; ok {
+			for _, builtInRoleName := range builtInRoleNames {
```

Comment:
Maybe we could move the content of the if in another function: `func appendBuildInRolesPermissions(permissions, builtInRoleNames) []*accesscontrol.Permission`. Your call, the function is not that big but has a lot of for/if/for/if/for encapsulated.